### ISSUE TITLE
Bug: iterator returned by `MidiMessage::bytes()` does not iterate

### ISSUE THREAD

**Issue by Janonard** [wmedrano/wmidi#6]
I want to read the byte-representation of a message byte-per-byte. This should usually be possible by calling `MidiMessage::bytes`, which is method provided by the `Read` trait. However, the returned iterator always yields the first byte of the message.

#### `Cargo.toml`

``` toml
[package]
name = "wmidi-bug"
version = "0.1.0"
authors = ["Jan-Oliver Janonard opdenhoevel <jan.opdenhoevel@protonmail.com>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
wmidi = "3.0.1"
```

#### `main.rs`

``` rust
extern crate wmidi;

use std::convert::TryFrom;
use std::io::Read;
use wmidi::*;

fn main() {
    let message = MidiMessage::NoteOn(Channel::Ch1, Note::C0, Velocity::try_from(127).unwrap());
    for read_val in message.bytes() {
        if let Ok(byte) = read_val {
            println!("{}", byte);
        } else {
            panic!("Read error!");
        }
    }
}
```

#### Output

```
144
144
144
... (infinite loop)
```

#### Expected Output

```
144
24
127

**Comment by wmedrano** [comment]
The problem is that the read implementation is meant for a one-shot into a `&mut[u8]`. It can't really be made into an iterator directly because it has no state of its own. The workaround would be to create a slice with size of at least `MidiMessage::wire_size()`, call `Read::read` to load data to the buffer and call `.iter()` on the buffer.

**Comment by wmedrano** [comment]
Longer term, maybe we want to do the following:
1. Prevent use of the `Bytes` method in the trait. Maybe stop implementing `Read` altogether? `Read` seems to imply stateful.
2. Provide a `to_bytes` method to remove the boilerplate required to get the bytes. For performance tuning, maybe the caller can choose to store the value in a statically sized array, a smallvec or a Vec.

Do you have any extra details on your usecase to help develop a clean API?

**Comment by Janonard** [comment]
TL;DR: I would suggest to drop the `Read` trait as well as the implementation of a `copy_to_slice` and a `raw_size` method.

I would suggest to drop the `Read` trait because it not only implies statefulness, but also the possibility to block the thread: Your implementation won't do that, of course, but the documentation of the trait says that an implementer *might*. This is something you don't want to have in a real-time-sensitive environment and might "scare off" users.

However, I think that the current implementation of `read` could be used as a `copy_to_slice`-method. This is exactly what it does.

I would also appreciate a `raw_size` method which returns the size of the message. I'm working on the midi crate for [rust-lv2](https://github.com/RustAudio/rust-lv2) and need to know the exact size of the message before writing it.

**Comment by wmedrano** [comment]
`MidiMessage::bytes_size()` and `MidiMessage::copy_to_slice` live in 3.1.0.

**Comment by wmedrano** [comment]
Will look into implementing an iterator after setting up benchmark suite that can run on travis.
